src/opm/io/eclipse/EclOutput.cpp

```diff
@@ -270,27 +268,56 @@ void EclOutput::writeBinaryArray(const std::vector<T>& data)
 
         ofileH.write(reinterpret_cast<char*>(&dhead), sizeof(dhead));
 
-        for (int i = 0; i < num; i++) {
-            if (arrType == INTE) {
-                rval = flipEndianInt(data[n]);
-                ofileH.write(reinterpret_cast<char*>(&rval), sizeof(rval));
-            } else if (arrType == REAL) {
-                value_f = flipEndianFloat(data[n]);
-                ofileH.write(reinterpret_cast<char*>(&value_f), sizeof(value_f));
-            } else if (arrType == DOUB) {
-                value_d = flipEndianDouble(data[n]);
-                ofileH.write(reinterpret_cast<char*>(&value_d), sizeof(value_d));
-            } else if (arrType == LOGI) {
-                intVal = data[n] ? logi_true_val : false_value;
-                ofileH.write(reinterpret_cast<char*>(&intVal), sizeOfElement);
-            } else {
-                std::cerr << "type not supported in write binaryarray\n";
-                std::exit(EXIT_FAILURE);
-            }
+        if (arrType == INTE) {
 
-            n++;
+            std::vector<int> flipped_data;
+            flipped_data.resize(num, 0);
+
+            for (int m = 0; m < num; m++)
+                flipped_data[m] = flipEndianInt(data[m + offset]);
+
+            ofileH.write((char*)(&flipped_data[0]), flipped_data.size() * sizeof(int)) ;
```

Comment:
Consider using `flipped_data.data()` instead of `&flipped_data[0] ` Applies to similar calls to the write function.
https://stackoverflow.com/a/28707981

Reply:
thanks, updated.